src/services/cbridge1.execute.ts

```diff
@@ -0,0 +1,121 @@
+import { constants } from 'ethers'
+
+import { ExecuteCrossParams, getChainById } from '../types'
+import { checkAllowance } from './allowance.execute'
+import cbridge1 from './cbridge1'
+import Lifi from './LIFI/Lifi'
+import notifications, { NotificationType } from './notifications'
+import { createAndPushProcess, initStatus, setStatusDone } from './status'
+import { personalizeStep } from './utils'
+
+export class Cbridge1ExecutionManager {
+  shouldContinue: boolean = true
+
+  setShouldContinue = (val: boolean) => {
+    this.shouldContinue = val
+  }
+
+  executeCross = async ({ signer, step, updateStatus }: ExecuteCrossParams) => {
+    // setup
+    const { action, execution, estimate } = step
+    const { status, update } = initStatus(updateStatus, execution)
+    const fromChain = getChainById(action.fromChainId)
+
+    // STEP 1: Check Allowance ////////////////////////////////////////////////
+    if (action.fromToken.id !== constants.AddressZero) {
+      // Check Token Approval only if fromToken is not the native token => no approval needed in that case
+      if (!this.shouldContinue) return status
+      await checkAllowance(
+        signer,
+        fromChain,
+        action.fromToken,
+        action.fromAmount,
+        estimate.approvalAddress,
+        update,
+        status,
+        true,
+      )
+    }
+
+    // STEP 2: Get Transaction ////////////////////////////////////////////////
+    const crossProcess = createAndPushProcess('crossProcess', update, status, 'Prepare Transaction')
+
+    const personalizedStep = await personalizeStep(signer, step)
+    const { tx: transactionRequest } = await Lifi.getStepTransaction(personalizedStep)
+
+    // STEP 3: Send Transaction ///////////////////////////////////////////////
+    crossProcess.status = 'ACTION_REQUIRED'
+    crossProcess.message = 'Sign Transaction'
+    update(status)
+    if (!this.shouldContinue) return status
+
+    const tx = await signer.sendTransaction(transactionRequest)
```

Comment:
Here the check if the transaction was already sent (`crossProcess.txHash` exists) is missing, right?